### PR TITLE
Add page view data to events

### PR DESCRIPTION
**Pull request by sanderpick**
This was a bit more involved than I originally thought, but I'm happy with the outcome.

- Adds all test variants to every event via new `VariantContext` provider.
- Fixes a regex bug that let to favoring the "unidentified user" variant in A/B tests
- Switches to random id generation from UUID V4, which may have been introducing some bias to the ConfigCat bucket selection: https://github.com/configcat/common-js/blob/master/src/RolloutEvaluator.ts#L335
- On all pages, don't use an image background in hero for small sizes. Makes text more legible.

### CODE REVIEW

File: lib/users.ts [hunk 1]
```diff
@@ -14,8 +14,7 @@ export function getUserId(): string | undefined {
   // Try to get from `set-cookie` header
   const header = headers().get("set-cookie")
   if (header) {
-    // Note: This pattern can be improved after https://github.com/vercel/next.js/issues/47189 is resolved.
-    const match = header.match(/^id=([a-f0-9,-]*);/i)
+    const match = header.match(/id=([a-z0-9]+);/i)
```
Comment (sanderpick) [inline lib/users.ts:17]: Here's the regex issue. The old regex worked before I gave the cookie an expiration data, which changes the structure of the set-cookie header.

File: middleware.ts [hunk 2]
```diff
@@ -21,3 +18,11 @@ export async function middleware(req: NextRequest) {
...
   return res
 }
+
+// Returns a random (not necessarily unique) id of length 21.
+function newId() {
+  return (
+    Math.random().toString(36).substring(2) +
+    Math.random().toString(36).substring(2)
+  )
+}
```
Comment (sanderpick) [inline middleware.ts:28]: Switch to random ids instead of UUID v4

File: tailwind.config.js [hunk 3]
```diff
@@ -15,7 +15,7 @@ module.exports = {
         neonblue: "#75b6b5",
         grey: "#5d657f",
         orange: "#f4706b",
-        pink: "#d96e75",
+        pink: "#da616d",
```
Comment (sanderpick) [inline tailwind.config.js:18]: Better pink.

File: app/pilot-program/page.tsx [hunk 4]
```diff
@@ -34,7 +34,7 @@ export default function PilotProgramPage() {
     <main>
       {/* <!-- hero --> */}
       <div className="bg-white">
-        <section className="w-full bg-top bg-cover bg-no-repeat bg-[url('/img/pilot/hero.jpg')]">
+        <section className="w-full bg-top bg-cover bg-no-repeat bg-white md:bg-[url('/img/pilot/hero.jpg')]">
```
Comment (sanderpick) [inline app/pilot-program/page.tsx:37]: On all pages, don't use an image background for small sizes. Makes text more legible.

**Review by dtbuchholz — APPROVED** [review]
fair trade! 🤝

**Pull request merged**